src/ts/features/batch-editing.ts

```diff
@@ -0,0 +1,115 @@
+import {Feature, Shortcut} from '../utils/settings'
+import {Roam} from '../roam/roam'
+import {getHighlightedBlocks} from '../utils/dom'
+import {forEachAsync, delay} from '../utils/async'
+
+export const config: Feature = {
+    id: 'editing',
+    name: 'Batch Editing',
+    settings: [
+        {
+            type: 'shortcut',
+            id: 'batchLinking',
+            label: 'Apply [[link]] brackets to a word in every highlighted block\n\n(whole-words only)',
+            initValue: 'Meta+shift+l',
+            onPress: () => batchLinking(),
+        } as Shortcut,
+        {
+            type: 'shortcut',
+            id: 'batchAppendTag',
+            label: 'Append #yourtag to every highlighted block',
+            initValue: 'Ctrl+shift+t',
+            onPress: () => appendTag(),
+        } as Shortcut,
+        {
+            type: 'shortcut',
+            id: 'removeLastEndTag',
+            label: 'Remove the last #tag at the end of each highlighted block',
+            initValue: 'Ctrl+shift+meta+t',
+            onPress: () => removeLastTag(),
+        } as Shortcut,
+        {
+            type: 'shortcut',
+            id: 'regexSearchAndReplace',
+            label:
+                'Roll your own complex search and replace by providing a search string or regex plus a replacement string',
+            initValue: 'Ctrl+shift+f',
+            onPress: () => regexSearchAndReplace(),
+        } as Shortcut,
+    ],
+}
+
+const batchLinking = () => {
+    const text = prompt('What (whole) word do you want to convert into bracketed links?')
+    if (!text || text === '') return
```

Comment:
second check is redundant `''` is "falsey"

Reply:
+ this for all other similar checks

Reply:
✅